### PR TITLE
Update SnpEff tool_data_tables and SnpEffDb to enable multiple versions of genomedb files.

### PR DESCRIPTION
**Pull request by jj-umn**
SnpEff v4.1 updates the genome builds again, and SnpEff depends on genome builds that are compatiebl with the version.  The snpeff .loc files have a key and (snpeff)version column added so that the tool snpEff can filter reference genome builds by snpeff version.

### CODE REVIEW

**Comment by jj-umn** [comment]
@bgruening  @nsoranzo  Does this pull request look OK?   I need to get this into a couple production workflows soon, and rather use an IUC version from toolshed than create my own.  
Thanks.

**Comment by bgruening** [comment]
Reviewing it right now :)
Moreover, the download to of snpeff data is running :)

**Comment by nsoranzo** [comment]
I have not been using SnpEff for some time and I'm not an expert of data managers, by just looking at the code seems ok, feel free to proceed.

File: tool_collections/snpeff/snpEff_macros.xml [hunk 1]
```diff
@@ -10,6 +10,9 @@
         <exit_code range="1:"  level="fatal" description="Error" />
     </stdio>
   </xml>
+  <token name="@WRAPPER_VERSION@">4.0</token>
+  <token name="@SNPEFF_VERSION@">SnpEff4.0</token>
+  <token name="@SNPEFF_DATABASE_URL@">http://sourceforge.net/projects/snpeff/files/databases/v4_0/</token>
```
Comment (bgruening) [inline tool_collections/snpeff/snpEff_macros.xml]: @jj-umn one former reference is pointing to `../databases/` without `v4_0`. I think `v4_0` is correct, just double checking.

File: tool_collections/snpeff/snpEff.xml [hunk 2]
```diff
@@ -115,27 +117,33 @@
...
                 </param>
             </when>
             <when value="history">
-                <param format="snpeffdbv4" name="snpeff_db" type="data" label="SnpEff Genome Version Data"/>
+                <param format="snpeffdb" name="snpeff_db" type="data" label="@SNPEFF_VERSION@ Genome Data">
+                    <options options_filter_attribute="metadata.snpeff_version" >
+                        <filter type="add_value" value="@SNPEFF_VERSION@" />
+                    </options>
+                    <validator type="expression" message="This version of SnpEff will only work with @SNPEFF_VERSION@ Genome databases.">value is not None and value.metadata.snpeff_version == "@SNPEFF_VERSION@"</validator>
+                </param>
```
Comment (bgruening) [inline tool_collections/snpeff/snpEff.xml]: This is just great!

**Comment by bgruening** [comment]
@jj-umn can you double check the SNPEFF_URL please. I will merge this now. Today the download is finished and we can change the URL. After this we are ready to put it into the toolsheds.

Thanks, this is really great!

 * Note to myself: Put the datatype into Galaxy

**Comment by jj-umn** [comment]
@bgruening  Pablo is versioning the databases, so I changed the SNPEFF_URL to point to the correct version for this snippet release.   I'll need to remember to update it when I update to version 4.1

**Pull request merged**